AK/String.h

```diff
@@ -51,8 +51,13 @@ class String : public Detail::StringBase {
     // Creates a new String from a sequence of UTF-8 encoded code points.
     static ErrorOr<String> from_utf8(StringView);
 
+    enum class WithBOMHandling {
+        Yes,
+        No,
+    };
+
     // Creates a new String using the replacement character for invalid bytes
-    [[nodiscard]] static String from_utf8_with_replacement_character(StringView);
+    [[nodiscard]] static String from_utf8_with_replacement_character(StringView, WithBOMHandling = WithBOMHandling::No);
```

Comment:
hmm I think it might help match the calling specs more if the default is Yes. Then the call sites would look like:
```c++
// Return the result of running UTF-8 decode on bytes.
return String::from_utf8_with_replacement_character(bytes);

// Return the result of running UTF-8 decode without BOM on bytes.
return String::from_utf8_with_replacement_character(bytes, WithBOMHandling::No);
```

Reply:
Sure that makes sense, will change. I suppose I had that as the default as I like to pretend the BOM doesn't exist :D